python/arcticdb/util/venv.py

```diff
@@ -214,4 +211,53 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         unset_config_int("VersionMap.ReloadInterval")
         del self.lib
-        del self.ac
\ No newline at end of file
+        del self.ac
+
+
+class CompatLibrary:
+    """
+    Responsible for creating and cleaning up a library (or multiple libraries) when writing compatibility tests.
+
+    Usually the library cleanup is managed by pytest fixtures, but it is hard to do so for compatibility tests because
+    they are responsible for maintaining their own arctic instances across several processes. As long as libraries
+    within the test are wrapped in a CompatLibrary cleanup will be dealt with.
+    """
+    def __init__(self, old_venv : Venv, uri : str, lib_names : Union[str, List[str]], create_with_current_version=False):
+        self.old_venv = old_venv
+        self.uri = uri
+        self.lib_names = lib_names if isinstance(lib_names, list) else [lib_names]
+        assert len(self.lib_names) > 0
+        self.create_with_current_version = create_with_current_version
+
+    def __enter__(self):
+        for lib_name in self.lib_names:
+            if self.create_with_current_version:
+                ac = Arctic(self.uri)
+                ac.create_library(lib_name)
+                del ac
+            else:
+                old_ac = self.old_venv.create_arctic(self.uri)
+                old_ac.create_library(lib_name)
+
+        self.old_ac = self.old_venv.create_arctic(self.uri)
+        self.old_libs = {lib_name : self.old_ac.get_library(lib_name) for lib_name in self.lib_names}
+        self.old_lib = self.old_libs[self.lib_names[0]]
+        return self
+
+    def current_version(self):
+        return CurrentVersion(self.uri, self.lib_names[0])
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        ac = Arctic(self.uri)
+        for lib_name in self.lib_names:
+            try:
+                ac.delete_library(lib_name)
+            except StorageException as e:
+                if self.uri.startswith("lmdb"):
+                    # For lmdb on windows sometimes the venv process keeps a dangling reference to the library for some
+                    # time. Since we use temporary directories for lmdb it's fine to not retry because the temporary
+                    # directory will be cleaned up in the end of the pytest process.
+                    logger.info(f"Failed to delete an lmdb library due to a Storage Exception: {e}")
+                else:
+                    raise
+        del ac
```

Comment:
Why do we need the `del ac` calls in this PR, especially this last one? Python gives very loose guarantees about what `del` actually does, so it's a bit of a worry (especially when the things we're calling `del` on are about to go out of scope anyway)